### PR TITLE
WIP: 0.6.0 prep

### PR DESCRIPTION
**Pull request by matzew**
moar OLM bits for the 0.6.0 edition

### CODE REVIEW

**Comment by matzew** [comment]
@jcrossley3 I see the serving counterpart has some nice scorecard thingy ... I guess should be done here too ?

**Comment by matzew** [comment]
@lberk updates to get ready for 0.6.0 (work in progress)

**Comment by jcrossley3** [comment]
> @jcrossley3 I see the serving counterpart has some nice scorecard thingy ... I guess should be done here too ?

Yeppers! Update the versions in `.osdk-scorecard.yaml` after you generate your new CSV's and run `operator scorecard`.

File: deploy/olm-catalog/knative-eventing-operator/0.6.0/eventing_v1alpha1_install_crd.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            namespace:
```
Comment (jcrossley3) [inline deploy/olm-catalog/knative-eventing-operator/0.6.0/eventing_v1alpha1_install_crd.yaml:34]: FYI, I'd like to remove `spec.namespace` and `status.resources`, but I'd like to merge this first. No sense in compounding changes in a single WIP. Unlike you, I do NOT prefer to drink from the firehose. ;)

File: deploy/olm-catalog/knative-eventing-operator/0.6.0/eventing_v1alpha1_install_crd.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+              description: The target namespace in which to install the upstream resources
+              type: string
+          type: object
+        status:
+          properties:
+            resources:
```
Comment (jcrossley3) [inline deploy/olm-catalog/knative-eventing-operator/0.6.0/eventing_v1alpha1_install_crd.yaml:40]: FYI, I intend to replace this with `status.conditions`

File: deploy/olm-catalog/knative-eventing-operator/0.6.0/knative-eventing-operator.v0.6.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.namespace
+                - name: POD_NAME
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.name
+                - name: OPERATOR_NAME
+                  value: knative-eventing-operator
+                image: quay.io/openshift-knative/knative-eventing-operator:v0.5.0
```
Comment (jcrossley3) [inline deploy/olm-catalog/knative-eventing-operator/0.6.0/knative-eventing-operator.v0.6.0.clusterserviceversion.yaml]: This should be 0.6.0, and should've carried over from `deploy/operator.yaml`, so I'm assuming you generated this _before_ updating `operator.yaml`?

File: deploy/olm-catalog/knative-eventing-operator/0.6.0/knative-eventing-operator.v0.6.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+                name: knative-eventing-operator
+            spec:
+              containers:
+              - command:
+                - knative-eventing-operator
+                env:
+                - name: WATCH_NAMESPACE
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.namespace
```
Comment (jcrossley3) [inline deploy/olm-catalog/knative-eventing-operator/0.6.0/knative-eventing-operator.v0.6.0.clusterserviceversion.yaml]: This should be `metadata.annotations['olm.targetNamespaces']` and `gen-csv` will hopefully generate that for you correctly some day.

**Review by jcrossley3 — CHANGES_REQUESTED** [review]
Looks good!

File: deploy/olm-catalog/knative-eventing-operator/0.6.0/knative-eventing-operator.v0.6.0.clusterserviceversion.yaml (hunk 3 above)
Comment (matzew) [reply deploy/olm-catalog/knative-eventing-operator/0.6.0/knative-eventing-operator.v0.6.0.clusterserviceversion.yaml]: @jcrossley3  yes - that's exactly right :-)   and now I fixed it

File: deploy/olm-catalog/knative-eventing-operator/0.6.0/knative-eventing-operator.v0.6.0.clusterserviceversion.yaml (hunk 4 above)
Comment (matzew) [reply deploy/olm-catalog/knative-eventing-operator/0.6.0/knative-eventing-operator.v0.6.0.clusterserviceversion.yaml]: @jcrossley3 ok, thanks for that pointer - I've fixed this

**Comment by matzew** [comment]
@jcrossley3 updated w/ the two low hanging fruit changes.


Missing, as indicated on your comments, those can be different PR:
* your suggestion to remove `spec.namespace and `status.resources` 
* usage of `status.conditions`

I saw some of these suggestion on the serving operator, but was not really 100% sure - that's also I why I marked this WIP, and figured we take it from there.

Thanks for looking!

**Review by jcrossley3 — APPROVED** [review]

**Pull request merged**